psi4/src/psi4/libfock/snLinK_stub.cc

```diff
@@ -0,0 +1,52 @@
+/*
+ * @BEGIN LICENSE
+ *
+ * Psi4: an open-source quantum chemistry software package
+ *
+ * Copyright (c) 2007-2023 The Psi4 Developers.
+ *
+ * The copyrights for code used from other parties are included in
+ * the corresponding files.
+ *
+ * This file is part of Psi4.
+ *
+ * Psi4 is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, version 3.
+ *
+ * Psi4 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along
+ * with Psi4; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * @END LICENSE
+ */
+
+#include "jk.h"
+#include "SplitJK.h"
+#include "psi4/libpsi4util/PsiOutStream.h"
+
+// "stub" implementation for snLinK when GauXC isn't installed. 
+// Just throws on calling the constructor.
+namespace psi {
+
+snLinK::snLinK(std::shared_ptr<BasisSet> primary, Options& options) : SplitJK(primary, options) {
```

Comment:
I am wondering if this stub file is necessary. Why can we not just throw this as a PSIEXCPTION in the main `snLinK.cc` file.

For example

```
#ifndef USING_gauxc
    throw PSIEXCEPTION("snLinK was requested in SCF_TYPE, but GauXC is not installed! Please recompile Psi4 with ENABLE_gauxc=ON.");
#endif
```

Reply:
It's not technically necessary, to be honest. I did it initially to avoid having to use macros in the main file, a sort of extra modularity. But I'm happy to just use the macro(s) approach, as well.

Reply:
Done!